File: php/features.php
<?php
declare(strict_types=1);

// Genderizeio SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class GenderizeioFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new GenderizeioBaseFeature();
            case "test":
                return new GenderizeioTestFeature();
            default:
                return new GenderizeioBaseFeature();
        }
    }
}

File: py/genderizeio_sdk/features.py
# Genderizeio SDK feature factory

from genderizeio_sdk.feature.base_feature import GenderizeioBaseFeature
from genderizeio_sdk.feature.test_feature import GenderizeioTestFeature


def _make_feature(name):
    features = {
        "base": lambda: GenderizeioBaseFeature(),
        "test": lambda: GenderizeioTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
